js/app/src/css.ts

```diff
@@ -18,3 +18,42 @@ export function mount_css(url: string, target: HTMLElement): Promise<void> {
 		target.appendChild(link);
 	});
 }
+
+export function prefix_css(
+	string: string,
+	version: string,
+	style_element = document.createElement("style")
+): HTMLStyleElement {
+	style_element.remove();
+
+	const stylesheet = new CSSStyleSheet();
+	stylesheet.replaceSync(string);
+
+	const rules = stylesheet.cssRules;
+
+	let css_string = "";
+
+	for (let i = 0; i < rules.length; i++) {
+		const rule = rules[i];
+
+		if (rule instanceof CSSStyleRule) {
+			const selector = rule.selectorText;
+			if (selector) {
+				const new_selector = selector
+					.split(",")
+					.map(
+						(s) =>
+							`gradio-app .gradio-container.gradio-container-${version} .contain ${s.trim()}`
+					)
+					.join(",");
+
+				css_string += rule.cssText;
+				css_string += rule.cssText.replace(selector, new_selector);
```

Comment:
Nice. I don't think repeating a css rule should have any (non-performance-related) downsides so this should be good

Reply:
The number of rules should generally be pretty modest, so I think the performance should be ok.